lib/api_repo.js

```diff
@@ -316,87 +316,80 @@ ApiRepo.prototype.buildCommonProtoPkgs =
  * @param {function} done the function to run on protoc completion
  */
 ApiRepo.prototype._buildProtos =
-  function _buildProtos(name, version, language, done) {
-    var that = this;
-    /** findOutputs lists the files in the output directory */
-    var findOutputs = function findOutputs(err) {
-      if (err) {
-        console.error('findOutputs:start:err', err);
-        done(err);
-        return;
-      }
-      var outputs = [];
-      var langRoot = path.join(that.outDir, language);
-      var finder = new FindFiles({
-        rootFolder : langRoot,
-        filterFunction : function (_unused, stat) {
-          return stat.isFile();
-        }
-      });
-      var finished = false;
-      finder.on("complete", function() {
-        if (!finished) {
-          done(null, outputs);
-          finished = true;
-        }
-      });
-      finder.on("match", function(path) {
-        var shortPath = path.replace(langRoot + '/', '');
-        outputs.push(shortPath);
-      });
-      finder.on("error", function(err) {
-        if (!finished) {
-          console.error('findOutputs:err', err);
+    function _buildProtos(name, version, language, done) {
+      var that = this;
+      var topDir = this.outDir;
+      var langTopDir = path.join(this.outDir, language);
+
+      /** findOutputs lists the files in the output directory */
+      var findOutputs = function findOutputs(err) {
+        if (err) {
+          console.error('findOutputs:start:err', err);
           done(err);
-          finished = true;
+        } else {
+          var stripRoot = function stripRoot(err, outputs) {
+            if (err) {
+              done(err);
+            } else {
+              var stripped = _.map(outputs, function(x) {
+                return x.replace(langTopDir, '');
+              });
+              done(null, stripped);
+            }
+          };
+          glob.glob("**", {
+            cwd: langTopDir,
+            nodir: true,
+          }, stripRoot);
         }
-      });
-      finder.startSearch();
+      }.bind(this);
+      if (language === 'java') {
+        var dstResourceDir = path.join(langTopDir, 'src', 'main', 'resources');
+        fs.mkdirsSync(dstResourceDir);
+        /**
+         * copyJavaPb copies a protocol buffer file to the java package resource
+         * folder.
+         */
+        var copyJavaPb = function copyJavaPb(protoPath, next) {
+          var src = path.join(this.repoDir, protoPath),
+          dst = path.join(dstResourceDir, protoPath);
+          fs.copy(src, dst, next);
+        }.bind(this);
+        this._findProtos(name, version, findOutputs, copyJavaPb);
+      } else if (language === 'nodejs') {
+        /**
+         * makeNodeModule writes a commonJS module containing all the protos
+         * used by service.
+         */
+        var makeNodeModule = function makeNodeModule(err, allProtos) {
```

Comment:
You can simply write `function makeNodeModule(err, allProtos)`, `var makeNodeModule` is not necessary.